snippets: add snippet for tfm logging on shared uart (#12488)
nrfconnect/sdk-nrf: doc/nrf/libraries/modem/nrf_modem_lib/nrf_modem_lib_trace.rst

To send modem traces over UART on an nRF91 Series DK, configuration must be added for the UART device in the devicetree and Kconfig.
This is done by adding the :ref:`modem trace UART snippet <nrf91_modem_trace_uart_snippet>` when building and programming.

> To enable TFM logging while enabling modem traces, it is important to have the TFM logging on another UART than the modem traces. To do so,
> the TFM logging need to utilize the same UART as the application. This can be done by enabling the ``tfm-enable-share-uart`` snippet.

Reviewer: ```suggestion
To enable TF-M logging while using modem traces, it is important to have the output on different UARTs.
TF-M logging must use the same UART as the application.
Use the ``tfm-enable-share-uart`` parameter for sharing the UART.
```

Use the `Cellular Monitor`_ app for capturing and analyzing modem traces.

To enable TF-M logging while using modem traces, it is important to have the output on different UARTs.

> TF-M logging must use the same UART as the application, more information on how this is done is given in :ref:`TF-M shared logging snippet <tfm_enable_share_uart>`.

Reviewer: ```suggestion
TF-M logging must use the same UART as the application.
For more details, see :ref:`shared TF-M logging <tfm_enable_share_uart>`.
```
